colander/__init__.py

```diff
@@ -2273,6 +2270,21 @@ def __init__(self, *args, **kw):
             raise Invalid(self,
                           'Sequence schemas must have exactly one child node')
 
+    def clone(self):
+        """ Clone the schema node and return the clone.  All subnodes
+        are also cloned recursively.  Attributes present in node
+        dictionaries are preserved."""
+
+        # Cloning a ``SequenceSchema`` doesn't work with ``_SchemaNode.clone``.
+
+        children = [node.clone() for node in self.children]
+        cloned = self.__class__(self.typ, *children)
+
+        attributes = self.__dict__.copy()
+        attributes.pop('children', None)
+        cloned.__dict__.update(attributes)
+        return cloned
```

Comment:
@tseaver : There already is ``test_clone_with_sequence_schema`` from b5d4207 / #212: https://github.com/Pylons/colander/blob/master/colander/tests/test_colander.py#L3426-L3431.  I extended this test with some additional assertions.  Let me know if you think that more is needed.